src/components/tree/tree-item.ts

```diff
@@ -172,6 +182,23 @@ export default class IgcTreeItemComponent extends EventEmitterMixin<
     this.init = false;
   }
 
+  public updated(): void {
```

Comment:
I am not sure if `updated` hook is the best place for this code, it is executed on each render and is not related to the children of the item.

It would be more appropriate to use `slotchange`, it is executed only for top level children changes, but I guess this is good enough.

If we want to listen for recursive children changes, we could use `MutationObserver`, but I think that would be too expensive.